src/features/FeatureBase.ts

```diff
@@ -0,0 +1,156 @@
+import { IFeature } from "./IFeature";
```

Comment:
This is doing the same as Feature.js. It allows us to make typescript features that behave the same way as the old js ones.
Eventually the Feature.js will be replaced with this when all Features are migrated to TypeScript.